src/test/java/com/williamfiset/algorithms/graphtheory/TravelingSalesmanProblemTest.java

```diff
@@ -11,58 +11,85 @@
 
   @Test(expected = IllegalArgumentException.class)
   public void testTspRecursiveInvalidStartNode() {
-    double[][] dist = {
-      {1, 2, 3},
-      {4, 5, 6},
-      {7, 8, 9}
-    };
-    new TspDynamicProgrammingRecursive(321, dist);
+    new TspDynamicProgrammingRecursive(321, new double[5][5]);
   }
 
   @Test(expected = IllegalArgumentException.class)
-  public void testTspIterativeInvalidStartNode() {
-    double[][] dist = {
-      {1, 2, 3},
-      {4, 5, 6},
-      {7, 8, 9}
-    };
-    new TspDynamicProgrammingIterative(321, dist);
+  public void testTspIterativeStartNodeTooLarge() {
+    new TspDynamicProgrammingIterative(321, new double[5][5]);
   }
 
-  @Test(expected = IllegalStateException.class)
+  @Test(expected = IllegalArgumentException.class)
+  public void testTspIterativeNegativeStartNode() {
+    new TspDynamicProgrammingIterative(-1, new double[5][5]);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
   public void testTspRecursiveNonSquareMatrix() {
```

Comment:
This test expected an IllegalStateException even though the code for checking non-square matrices throws an IllegalArgumentException, but the test still succeeded because the dist matrix was 2x3 and threw an IllegalStateException because N <= 2.